### DISCUSSION TITLE
24.05.20 Progress Meeting

### DISCUSSION THREAD

**Discussion by Woolfrey** [Woolfrey/software_robot_library#115] [Meetings]
#### Chair:
@Woolfrey

#### Attendees:
@Woolfrey 
@ssutjipto 
_____

## Announcements

- We're organizing to bring Shoudong to Leeds for a research collaboration. And he his expecting me to write control code for the robot. @ssutjipto can be the point of contact in Sydney.

## Agenda
_Note: Last week's actions become this week's agenda_

| Item | Responsible | Issue | Outcome |
|-------|------------------|----------|--------------|
| Create class member for joint damping vector | @ssutjipto | N/A | Done  by @Woolfrey and @ssutjipto simultaneously? |
| Continue ActionServer base class | @Woolfrey | [Link](https://github.com/Woolfrey/server_serial_link/issues/1) | Working on `SerialKinematicControl` and accompanying test script |

## Additional Discussion
| Topic | Raised By | Issue | Outcome |
|---------|---------------|---------|--------------|
Is the redundant control optimisation in the right order for the kinematic control class? | @ssutjipto | N/A | Order of arguments is incorrect, but it should be caught in compile or runtime. |
| In the track joint trajectory class, should first be solving for acceleration then compute torque | @Woolfrey | N/A | Put a comment in the code for @ssutjipto |
| It should be possible solve both acceleration & torque limits for joint trajectory tracking as a QP problem | @Woolfrey | #118 | Not urgent, but would be nice to do. |

## Actions
| Action | Responsible | Tag |
|----------|------------------|-------|
| Finish the test script for velocity level joint trajectory tracking | @Woolfrey | #116  |
| Continue on action server | @Woolfrey | [Link](https://github.com/Woolfrey/server_serial_link/issues/1)  |
| Type up math for joint level feedback in discussion below | @Woolfrey | N/A |
| Continue working on dynamic control class | @ssutjipto | #117 |

**Comment by Woolfrey** [comment]
@ssutjipto  here's the math for the trajectory tracking control problem:
```math
\boldsymbol{\tau} = \mathbf{M\ddot{q} + C\dot{q} + g}
```
where:
```math
\mathbf{\ddot{q} = \ddot{q}_\mathrm{d} + K_\mathrm{d}\left(\dot{q}_\mathrm{d} - \dot{q}\right) + K_\mathrm{p}\left(q_\mathrm{d} - q\right)}
```
versus:
```math
\boldsymbol{\tau} = \mathbf{M\ddot{q}_\mathrm{d}  + K_\mathrm{d}\left(\dot{q}_\mathrm{d} - \dot{q}\right) + K_\mathrm{p}\left(q_\mathrm{d} - q\right) + C\dot{q} + g}
```
In the latter case, if the gain matrices are diagonal, you will get poor trajectory tracking because of the dynamic coupling between the joints.

**Reply by ssutjipto** [reply]
I've corrected it now. I had placed and removed brackets in the wrong place.
